format.md

```diff
@@ -130,6 +130,64 @@ is a list of xy vertices. The child name of the outer list should be "polygons";
+    | ------- | ------------------ | ---------------------- |
+    | 1       | Point              | `"Point"`              |
+    | 2       | LineString         | `"LineString"`         |
+    | 3       | Polygon            | `"Polygon"`            |
+    | 4       | MultiPoint         | `"MultiPoint"`         |
+    | 5       | MultiLineString    | `"MultiLineString"`    |
+    | 6       | MultiPolygon       | `"MultiPolygon"`       |
+    | 11      | Point Z            | `"Point Z"`            |
+    | 12      | LineString Z       | `"LineString Z"`       |
+    | 13      | Polygon Z          | `"Polygon Z"`          |
+    | 14      | MultiPoint Z       | `"MultiPoint Z"`       |
+    | 15      | MultiLineString Z  | `"MultiLineString Z"`  |
+    | 16      | MultiPolygon Z     | `"MultiPolygon Z"`     |
+    | 21      | Point M            | `"Point M"`            |
+    | 22      | LineString M       | `"LineString M"`       |
+    | 23      | Polygon M          | `"Polygon M"`          |
+    | 24      | MultiPoint M       | `"MultiPoint M"`       |
+    | 25      | MultiLineString M  | `"MultiLineString M"`  |
+    | 26      | MultiPolygon M     | `"MultiPolygon M"`     |
+    | 31      | Point ZM           | `"Point ZM"`           |
+    | 32      | LineString ZM      | `"LineString ZM"`      |
+    | 33      | Polygon ZM         | `"Polygon ZM"`         |
+    | 34      | MultiPoint ZM      | `"MultiPoint ZM"`      |
+    | 35      | MultiLineString ZM | `"MultiLineString ZM"` |
+    | 36      | MultiPolygon ZM    | `"MultiPolygon ZM"`    |
+
+    These type id values were chosen to match the WKB specification exactly for 2D geometries and match the WKB specification conceptually for Z, M, and ZM geometries, given the constraint that an Arrow union type ID must be between 0 and 127.
+
+- A geometry array does not need to contain _all_ possible children arrays, but those children arrays must have the type ids defined above.
+- All children arrays of a geometry array must have the same dimensionality. So `Point` and `Polygon` arrays may be combined in a geometry array, but `Point` and `Polygon Z` arrays may not be.
+- The children arrays should not themselves contain GeoArrow metadata. Only the top-level geometry array should contain GeoArrow metadata.
+
+Note that single and multi geometries of the same type can be stored together in
+a Multi encoding without using this geometry type. For example, a mix of Polygon
+and MultiPolygon can be stored as MultiPolygons, with a Polygon being
+represented as a length-1 MultiPolygon. This is recommended over a geometry
+array if possible because it has less overhead per geometry.
+
+**GeometryCollection**: `List<Geometry>`
+
+An array of GeometryCollections is represented as a list containing the above
+geometry array. Each element of the array thus represents one or more geometries
+of varied type. The child name of the outer list should be "geometries".
+
```

Comment:
Should we note here that this means that "nested" geometry collections are therefore not supported, but those can always be represented in a flattened way?

Reply:
In [`ae6886c` (#43)](https://github.com/geoarrow/geoarrow/pull/43/commits/ae6886ce545cf3195e510909256c255c4ddde0b8) I explicitly specified that nested geometry collections are not supported.